src/inc/til/u8u16convert.h

```diff
@@ -306,25 +91,160 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
     // - E_OUTOFMEMORY - the function failed to allocate memory for the resulting string
     // - E_ABORT       - the resulting string length would exceed the upper boundary of an int and thus, the conversion was aborted before the conversion has been completed
     // - E_UNEXPECTED  - an unexpected error occurred
-    template<class inT, class outT>
-    [[nodiscard]] typename std::enable_if<std::is_same<typename inT::value_type, char>::value && std::is_same<typename outT::value_type, wchar_t>::value, HRESULT>::type
-    u8u16(const inT in, outT& out, u8state& state) noexcept
+    template<class outT>
+    [[nodiscard]] HRESULT u8u16(const std::string_view& in, outT& out, u8state& state) noexcept
     {
-        std::string_view sv{};
-        RETURN_IF_FAILED(state(std::string_view{ in }, sv));
-        return til::u8u16(sv, out);
+        try
+        {
+            /* clang-format off */
+            enum Utf8BitMasks : char
+            {
+                IsAscii   = gsl::narrow_cast<char>(0b0'0000000), // Any byte representing an ASCII character has the MSB set to 0
+                MaskAscii = gsl::narrow_cast<char>(0b1'0000000), // Bit mask to be used in a bitwise AND operation to find out whether or not a byte match the IsAscii pattern
+                IsCont    = gsl::narrow_cast<char>(0b10'000000), // Continuation bytes of any UTF-8 non-ASCII character have the MSB set to 1 and the adjacent bit set to 0
+                MaskCont  = gsl::narrow_cast<char>(0b11'000000), // Bit mask to be used in a bitwise AND operation to find out whether or not a byte match the IsCont pattern
+                IsLead2   = gsl::narrow_cast<char>(0b110'00000), // A lead byte that indicates a UTF-8 non-ASCII character consisting of two bytes has the two highest bits set to 1 and the adjacent bit set to 0
+                MaskLead2 = gsl::narrow_cast<char>(0b111'00000), // Bit mask to be used in a bitwise AND operation to find out whether or not a lead byte match the IsLead2 pattern
+                IsLead3   = gsl::narrow_cast<char>(0b1110'0000), // A lead byte that indicates a UTF-8 non-ASCII character consisting of three bytes has the three highest bits set to 1 and the adjacent bit set to 0
+                MaskLead3 = gsl::narrow_cast<char>(0b1111'0000), // Bit mask to be used in a bitwise AND operation to find out whether or not a lead byte match the IsLead3 pattern
+                IsLead4   = gsl::narrow_cast<char>(0b11110'000), // A lead byte that indicates a UTF-8 non-ASCII character consisting of four bytes has the four highest bits set to 1 and the adjacent bit set to 0
+                MaskLead4 = gsl::narrow_cast<char>(0b11111'000), // Bit mask to be used in a bitwise AND operation to find out whether or not a lead byte match the IsLead4 pattern
+            };
+            /* clang-format on */
+
+            constexpr const std::array<char, 4> cmpMasks{

```

Comment:
`constexpr` implies `const`. But you should make them `static` otherwise a copy is made every time the function is called.